Corrector

Ritu-Devnani opened the pull request:
Please review the files uploaded for training the corrector.

Ritu-Devnani committed:
Create train_corrector.py

Training the corrector requires call to the command line tools provided by fairseq. The definition of the above functions used is further provided in the utility file. These functions require the important parameters which are appended to the command written for calling the inbuilt fairseq tools. The training process requires good code ( correct python code ) and respective bad code.

Ritu-Devnani committed:
Create fairseq_utility.py

It contains function definitions used in train_corrector.py

akanshak984 commented on fairseq_utility.py:
```diff
@@ -0,0 +1,78 @@
+import shlex
+import shutil
```
Where have we used `shutil` in this file? If not let's remove this line

Ritu-Devnani replied:
shutil is used as a pre requisite as many files are generated after firing a commands related to fairseq. This library makes files copy, create, etc operations easier and faster.

Ritu-Devnani replied:
shutil is used as a pre requisite as many files are generated after firing a commands related to fairseq. This library makes files copy, create, etc operations easier and faster.

akanshak984 commented on fairseq_utility.py:
```diff
@@ -0,0 +1,78 @@
+                   --reset-meters "
+        cmd += additional_cmds
+        if logfile is not None:
+            import socket, os
```
We have imported os in the beginning of the file. Do we need to import it here again?

Ritu-Devnani replied:
Done the required changes.

akanshak984 commented on fairseq_utility.py:
```diff
@@ -0,0 +1,78 @@
+import shlex
+import shutil
+import subprocess
+import json, os
```
Have we used `json` anywhere?

Ritu-Devnani replied:
json is required afterwards for the functions defined for evaluation for easy understanding. I will remove that for now.

akanshak984 commented on fairseq_utility.py:
```diff
@@ -0,0 +1,78 @@
+import subprocess
+import json, os
+from pathlib import Path
+from collections import defaultdict, OrderedDict
```
Please verify all the imports and if it's not used let's remove it.

Ritu-Devnani replied:
subprocess : used for visualising the child processes running while a call to the fairseq_train() function. 
json : Initially removed
os : used for accessing the working directories and retrieve the id of the processes that are running ( this helps us to see the total number of tokens visited, etc. )
Path : used to indicate that a google drive path to which we need to store the output files generated.
collections : the token vocabulary taken as an input through parameter in the above function defined ( "srcdict" ) is stored and used in thee form of a dictionary.

Ritu-Devnani committed:
Update fairseq_utility.py

Removed duplicate import of os.

Ritu-Devnani committed:
Update fairseq_utility.py

I have removed the json library for now. It will be required while the evaluation functions are needed and defined.

akanshak984 commented:
Avoid commit messages like 
- *required changes* - It gives no idea of what required changes. specify exact changes if it is short or else summarize the changes.
- *thanks* or anything which doesn't give idea of what changes were done in the commit

example:
```
Update fairseq_utility.py
I have done the required changes. Thanks for pointing it out.
```

It gives no idea of what required changes are done.

instead you could write:

```
Update fairseq_utility.py

Remove duplicate import of os

```

akanshak984 commented:
Update the commit message and then we can merge this PR
